### PR TITLE
[ReadyToBeMerged][AnnotateThrowables] Support `$this` calling a method of the same class

### PR DESCRIPTION
**Pull request by Aerendir**

### CODE REVIEW

**Comment by Aerendir** [comment]
@TomasVotruba , ready to be merged 💪

File: src/PhpParser/Node/Value/ClassResolver.php [hunk 1]
```diff
@@ -54,37 +54,20 @@ private function resolveFromExpression(Expression $expression): ?FullyQualified
...
 
-    private function tryToResolveClassMethodStmts(ClassMethod $classMethod): ?FullyQualified
+    private function tryToResolveClassMethodFromThis(ClassMethod $classMethod): ?FullyQualified
     {
-        // $ this -> method();
-        $stmts = $classMethod->stmts;
-        if ($stmts === null) {
-            return null;
-        }
+        $class = $classMethod->name->getAttribute(ClassLike::class)->name;
```
Comment (TomasVotruba) [inline src/PhpParser/Node/Value/ClassResolver.php]:
```suggestion
        $class = $classMethod->name->getAttribute(AttributeKey::CLASS_NAME);
```

File: src/PhpParser/Node/Value/ClassResolver.php [hunk 2]
```diff
@@ -54,37 +54,20 @@ private function resolveFromExpression(Expression $expression): ?FullyQualified
...
     private function resolveFromClassMethod(ClassMethod $classMethod, MethodCall $methodCall): ?FullyQualified
     {
-        $class = $this->tryToResolveClassMethodStmts($classMethod);
-
-        if ($class === null) {
-            $class = $this->tryToResolveClassMethodParams($classMethod, $methodCall);
-        }
-
-        return $class;
+        return 'this' === $methodCall->var->name
```
Comment (TomasVotruba) [inline src/PhpParser/Node/Value/ClassResolver.php]:
```suggestion
        return $this->isName($methodCall->var, 'this')
```

File: src/PhpParser/Node/Value/ClassResolver.php [hunk 3]
```diff
@@ -53,32 +53,25 @@ private function resolveFromExpression(Expression $expression): ?FullyQualified
...
-
-        if ($class === null) {
-            $class = $this->tryToResolveClassMethodParams($classMethod, $methodCall);
+        $var = $methodCall->var;
+        if (! $var instanceof Variable) {
+            return null;
         }
 
-        return $class;
+        return $var->name === 'this'
```
Comment (TomasVotruba) [inline src/PhpParser/Node/Value/ClassResolver.php]:
```suggestion
        return $this->isName($var, 'this')
```

File: src/PhpParser/Node/Value/ClassResolver.php (hunk 1 above)
Comment (Aerendir) [reply src/PhpParser/Node/Value/ClassResolver.php]: This doesn't work. it is not the same thing.

File: src/PhpParser/Node/Value/ClassResolver.php (hunk 2 above)
Comment (Aerendir) [reply src/PhpParser/Node/Value/ClassResolver.php]: Done

**Comment by Aerendir** [comment]
@TomasVotruba , I did the changes that was possible to make...

**Comment by TomasVotruba** [comment]
Thank you

**Pull request merged**